### ISSUE TITLE
`not-found` path doesn't include / derivate layout

### ISSUE THREAD

**Issue by sozonome** [vercel/next.js#52937]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.14.0
      npm: 9.7.2
      Yarn: 1.22.19
      pnpm: 8.3.0
    Relevant Packages:
      next: 13.4.10
      eslint-config-next: 13.4.10
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router

### Link to the code that reproduces this issue or a replay of the bug

https://vercel.com/sozonome/nextjs-reproduce-not-found-issue

### Steps to Reproduce

1. Install next v13.4.10 or later (including canary version)
2. Add `not-found.js/ts` file
3. Run `next dev` (also reproduced in build version)
4. Open any pathname that are not defined (that should produce 404 / not-found error)

### Actual Result

Will show 404 / not-found page, but doesn't include what's defined/composed in layout.js/ts

### Expected Result

Will show 404 / not-found page and include what's defined/composed in layout.js/ts

### Describe the Bug

`app/not-found` doesn't include layout since v13.4.10

not-found not include layout samples:
- next v13.4.10: 
  - commit: https://github.com/sozonome/nextjs-reproduce-not-found-issue/commit/640c5f6d57d3d7be89e93958f5516c564c94384c#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R29
  - deployment: https://nextjs-reproduce-not-found-issue-925964yy0-sozonome.vercel.app/aa  
- next v13.4.11-canary.0: 
  - commit: https://github.com/sozonome/nextjs-reproduce-not-found-issue/commit/3f7e9819abbb6cc89fc212b4d2e5ed8f06558510#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R29
  - deployment: https://nextjs-reproduce-not-found-issue-kyjpm7pnw-sozonome.vercel.app/aa

<img width="721" alt="image" src="https://github.com/vercel/next.js/assets/17046154/300f97d9-e9fd-4107-8251-9499f49bb111">

### Expected Behavior

`app/not-found` include layout (it has been that way before v13.4.10, and there's no update in the documentation of next.js that says otherwise)

- next v13.4.9 not-found sample (in this version, not-found still include layout): 
  - commit: https://github.com/sozonome/nextjs-reproduce-not-found-issue/commit/82c3ca19f3488904aa673a7ab1fa5d2a34151636#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R29
  - deployment: https://nextjs-reproduce-not-found-issue-o82nnjrwr-sozonome.vercel.app/aa

<img width="1441" alt="Screenshot 2023-07-20 at 19 40 32" src="https://github.com/vercel/next.js/assets/17046154/2679b8e8-64c6-4fd2-8875-84980ea98566">

### Which browser are you using? (if relevant)

Brave v1.52.130

### How are you deploying your application? (if relevant)

Vercel

**Comment by huozhi** [comment]
Fixed by #52790

**Comment by sozonome** [comment]
It's reverted before release v13.4.11 ?
https://github.com/vercel/next.js/pull/52977

**Comment by Galanthus** [comment]
> Fixed by #52790

Sorry to say, but actually it's not fixed with the latest version.

**Comment by karlhorky** [comment]
It looks like @huozhi re-landed the fix in #52985 (reviewed by @ijjk and published in [`next@13.4.12`](https://github.com/vercel/next.js/releases/tag/v13.4.12))

**Comment by Galanthus** [comment]
> It looks like @huozhi re-landed the fix in #52985 (reviewed by @ijjk and published in [`next@13.4.12`](https://github.com/vercel/next.js/releases/tag/v13.4.12))

Having a custom not-fount.tsx/jsx won’t load id notFound is fired. This works fine in 13.4.9

**Comment by sozonome** [comment]
I think it's already fixed in next@13.4.12

**Comment by Galanthus** [comment]
> I think it's already fixed in next@13.4.12

that's what you Think.. but it is not. Easy closing?
